File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.go
func findMin(nums []int) int {
	left := 0
	right := len(nums) - 1

	for left < right {
		midpoint := left + (right -  left) / 2
		
		midValue := nums[midpoint]
		rightValue := nums[right]

		if midValue < rightValue {
			right = midpoint
		} else {
			left = midpoint + 1
		}
	}

	return nums[left]
}

File: Data Structures & Algorithms/search-2d-matrix/submission-0.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        flattened = []
        for row in matrix: 
            flattened += row

        left = 0
        right = len(flattened) - 1
        while left <= right:
            center = left + (right - left) // 2
            midpoint = flattened[center]
            if midpoint < target: 
                left = center + 1
            elif midpoint > target: 
                right = center - 1
            else:
                return True
        
        return False
